pandas/core/indexes/multi.py

```diff
@@ -1231,13 +1231,17 @@ def _format_native_types(self, na_rep="nan", **kwargs):
 
     def format(
         self,
+        name=None,
+        formatter=None,
+        na_rep=None,
+        names=False,
         space=2,
         sparsify=None,
         adjoin=True,
-        names=False,
-        na_rep=None,
-        formatter=None,
-    ):
+    ) -> list:
```

Comment:
Can you use `List` from typing and add a subscript?

Reply:
This has return type ``List[Union[str, Tuple[str]]]``.  It returns ``List[Tuple[str]]`` if adjoin=False, else ``List[str]``.

I'm not a fan of union return types, it can trip mypy up in random places. I would prefer to use Literal types and typing.cast, but that is only available in python 3.8. Hope that's ok?

Reply:
Sparsify is actually of type ``Union[bool, None, Literal[lib.no_default]]``, which can't be represented until python 3.8. I'll just leave it as is.

Reply:
I think should use the union or maybe even overload based off of your comment (@simonjayhawkins has used overload successfully in a few places already); makes for much better type checking than a generic list

Reply:
I agree, we don't use a bare list as typing *anywhere* else, I would change this (and use a Union)